### ISSUE TITLE
private endpoint in data resource ibm_secrets_manager_secret is incorrect

### ISSUE THREAD

**Issue by argeiger** [IBM-Cloud/terraform-provider-ibm#4187]
### Terraform CLI and Terraform IBM Provider Version

```
Terraform v1.2.9
on darwin_amd64
+ provider registry.terraform.io/ibm-cloud/ibm v1.48.0-beta0
```

### Affected Resource(s)

https://registry.terraform.io/providers/IBM-Cloud/ibm/latest/docs/data-sources/secrets_manager_secret

* ibm_secrets_manager_secret

### Terraform Configuration Files

```hcl
provider "ibm" {
  # Configuration options
  visibility = "private"
  region     = "ca-tor"
}

data "ibm_secrets_manager_secret" "secrets_manager_secret" {
    instance_id = "<INSTANCE ID>"
    secret_type = "arbitrary"
    secret_id = "<SECRET ID>"
}

```

### Debug Output

When `visibility = "private"`, it tries to access private.resource-controller.cloud.ibm.com which does not exist

<img width="1109" alt="image" src="https://user-images.githubusercontent.com/14077904/202751359-1206a013-2824-48d4-b386-0f4593600cc5.png">

### Expected Behavior

resource controller is only in us-south and us-east for private and should have accessed its appropriate endpoint
 
private.us-east.resource-controller.cloud.ibm.com
private.us-south.resource-controller.cloud.ibm.com

### Actual Behavior

Goes to wrong endpoint for resource-controller

**Comment by IdanAdar** [comment]
I do not believe this is specific to the Secrets Manager data resource. Additionally, new data sources were shipped. I suggest to try with them just in case. If the issue continues, please re-open.
